lib/mayaUsd/fileio/utils/meshReadUtils.cpp

```diff
@@ -657,6 +660,38 @@ UsdMayaMeshReadUtils::assignPrimvarsToMesh(const UsdGeomMesh& mesh,
                    typeName == SdfValueTypeNames->Color3fArray ||
                    typeName == SdfValueTypeNames->Float4Array ||
                    typeName == SdfValueTypeNames->Color4fArray) {
+            if (typeName == SdfValueTypeNames->FloatArray
+                && name != UsdMayaMeshColorSetTokens->DisplayOpacityColorSetName) {
+                // Could be anything, forcing us to guess...
+                std::string lowerCaseName(name.GetString());
+                std::transform(
+                    lowerCaseName.begin(),
+                    lowerCaseName.end(),
+                    lowerCaseName.begin(),
+                    [](unsigned char c) { return std::tolower(c); });
+                if (lowerCaseName.find("color") == std::string::npos
```

Comment:
That is an incredibly fragile test. Any suggestion for better filtering would be appreciated.